pkg/kubelet/cloud_request_manager.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"github.com/golang/glog"
+)
+
+type cloudResourceSyncManager struct {
+	// Cloud provider interface.
+	cloud cloudprovider.Interface
+	// Sync period
+	syncPeriod time.Duration
+
+	// To keep exclusive access to the requestParallelism
+	requestMux sync.Mutex
+	// Keep the count of requests processed in parallel (expected to be 1 at most at a given time)
+	requestParallelism chan int
+	// Sync with finished requests
+	requestSync chan int
+	// Request timeout
+	requestTimeout time.Duration
+	// To keep exclusive access to last node address and last err
+
+	lastNodeAddressesMux sync.Mutex
+	// Last list of node addresses retrieved from the cloud provider
+	lastNodeAddresses []v1.NodeAddress
+	// Last error retrieved from the cloud provider
+	lastNodeAddressError error
```

Comment:
I don't think we need any of the lastNodeAddress stuff now that we have nodeAddress set async from the from get and it is mutex protected.

Reply:
I removed the original internal go routine completely. It does not make sense to timeout given the request is running in its own go routine as part of the cloud request manager. So basically we are back with the original lines of code. Just wearing a different T-shirt.